bob/src/hw_counter.rs

```diff
@@ -0,0 +1,163 @@
+use crate::prelude::*;
+use bob_common::metrics::{
+    AMOUNT_DESCRIPTORS, CPU_LOAD, FREE_RAM, FREE_SPACE, TOTAL_RAM, TOTAL_SPACE,
+};
+use std::path::{Path, PathBuf};
+use sysinfo::{DiskExt, ProcessExt, System, SystemExt};
+
+const DESCRS_DIR: &str = "/proc/self/fd/";
+
+pub(crate) struct HWCounter {
+    disks: HashMap<PathBuf, String>,
+    interval_time: Duration,
+}
+
+impl HWCounter {
+    pub(crate) fn new(mapper: Arc<Virtual>, interval_time: Duration) -> Self {
+        let disks = Self::collect_used_disks(mapper.local_disks());
+        Self {
+            disks,
+            interval_time,
+        }
+    }
+
+    fn collect_used_disks(disks: &[DiskPath]) -> HashMap<PathBuf, String> {
+        System::new_all()
+            .disks()
+            .iter()
+            .filter_map(|d| {
+                let path = d.mount_point();
+                disks
+                    .iter()
+                    .find(move |dp| {
+                        let dpath = Path::new(dp.path());
```

Comment:
DiskPath::path() returns Path, isn't it?

Reply:
[it isn't.](https://github.com/qoollo/bob/blob/master/bob-common/src/data.rs#L270)